db/migrations/postgres/000105_add_namespace_and_data_id_to_blobs.up.sql

```diff
@@ -0,0 +1,25 @@
+BEGIN;
+
+CREATE TABLE temp_blobs (
+  seq            SERIAL          PRIMARY KEY,
+  namespace      VARCHAR(64)     NOT NULL,
+  hash           CHAR(64)        NOT NULL,
+  payload_ref    VARCHAR(1024)   NOT NULL,
+  created        BIGINT          NOT NULL,
+  peer           VARCHAR(256)    NOT NULL,
+  size           BIGINT,
+  data_id        UUID            NOT NULL
+);
+INSERT INTO temp_blobs (namespace, data_id, hash, payload_ref, created, peer, size)
+  SELECT DISTINCT data.namespace, data.id, data.blob_hash, blobs.payload_ref, blobs.created, blobs.peer, blobs.size
+  FROM data
+  LEFT JOIN blobs ON blobs.hash = data.blob_hash
+  WHERE data.blob_hash IS NOT NULL
+;
+DROP INDEX blobs_hash;
+DROP TABLE blobs;
+ALTER TABLE temp_blobs RENAME TO blobs; 
+CREATE INDEX blobs_hash_data_id ON blobs(hash, data_id);
```

Comment:
Should this just be `namespace, data_id`? Not sure the `hash` adds value, and particularly not first as `data_id` is the thing we expect to be unique in most practical cases.

Reply:
I thought we had discussed this and decided that we should always be looking up blobs by both their hash and data ID to prevent spoofing of different data?

Reply:
Yes, but I don't think that changes my suggestion on the index that's most efficient.

I believe the lookup is:
`{ ns: "mynamespace", data_id: "id12345", hash: "abce1234" }`

I assert the most efficient index to support this lookup is:
`namespace`, `data_id`

- Leaving off `namespace` feels wrong
- Adding in a 3rd dimension of `hash` seems unnecessary (as the scan length should be trivially small as we're covering a very minor edge case there)